astropy/utils/data.py

```diff
@@ -2046,6 +2046,11 @@ def _rmtree(path, replace=None):
                 )
             )
             raise
+        except OSError as e:
+            if e.errno == errno.EXDEV:
+                warn(e.strerror, AstropyWarning)
+                shutil.move(path, os.path.join(d, "to-zap"))
```

Comment:
Do we need this?
```suggestion
                shutil.move(path, os.path.join(d, "to-zap"))
            else:
                raise
```

Reply:
Can you clarify what you mean by "this"? Are you referring to the warning?

Reply:
When OSError is encountered, you caught it but only do something for `errno.EXDEV`, so that means currently, it silently ignores any OSError with other errno, which is undesirable, right? Or did I misunderstand?

Reply:
you are absolutely right! thanks..